packages/quantic/force-app/main/default/lwc/quanticFacet/__tests__/quanticFacet.test.js

```diff
@@ -231,9 +333,674 @@ describe('c-quantic-facet', () => {
+
+                const facetValuesShowLess = element.shadowRoot.querySelector(
+                  selectors.facetValuesShowLess
+                );
+                expect(facetValuesShowLess)[
+                  canShowLessValues ? 'toBeTruthy' : 'toBeNull'
+                ]();
+              });
+            });
+
+            it('should call the controller function showLessValues when clicked', async () => {
+              facetState = {
+                ...initialFacetState,
+                values: exampleFacetValues,
+                canShowLessValues: true,
+              };
+              const element = createTestComponent({
+                ...defaultOptions,
+                noSearch: true,
+              });
+              await flushPromises();
+
+              const facetValuesShowLess = element.shadowRoot.querySelector(
+                selectors.facetValuesShowLess
+              );
+              facetValuesShowLess.click();
+              await flushPromises();
+
+              expect(functionsMocks.showLessValues).toHaveBeenCalledTimes(1);
+            });
+          });
+
+          describe('the show more facet values button', () => {
+            [true, false].forEach((canShowMoreValues) => {
+              it(`should ${canShowMoreValues ? '' : 'not '}display show more facet values button when the state indicates that it is ${canShowMoreValues ? 'possible' : 'not possible'}`, async () => {
+                facetState = {
+                  ...initialFacetState,
+                  values: exampleFacetValues,
+                  canShowMoreValues,
+                };
+                const element = createTestComponent({
+                  ...defaultOptions,
+                  noSearch: true,
+                });
+                await flushPromises();
+
+                const facetValuesShowMore = element.shadowRoot.querySelector(
+                  selectors.facetValuesShowMore
+                );
+                expect(facetValuesShowMore)[
+                  canShowMoreValues ? 'toBeTruthy' : 'toBeNull'
+                ]();
+              });
+            });
+
+            it('should call the controller function showMoreValues when clicked', async () => {
+              facetState = {
+                ...initialFacetState,
+                values: exampleFacetValues,
+                canShowMoreValues: true,
+              };
+              const element = createTestComponent({
+                ...defaultOptions,
+                noSearch: true,
+              });
+              await flushPromises();
+
+              const facetValuesShowMore = element.shadowRoot.querySelector(
+                selectors.facetValuesShowMore
+              );
+              facetValuesShowMore.click();
+              await flushPromises();
+
+              expect(functionsMocks.showMoreValues).toHaveBeenCalledTimes(1);
+            });
+          });
+
+          ['checkbox', 'link'].forEach((propertyValue) => {
```

Comment:
this is also a test that appears twice, is the behaviour really impacted by `noSearch` being true?

Reply:
Yes! cause if you see the HTML template, the case where the facet search is activated and the case where it not are handled differently. so I needed to test each case separetly.